src/xngin/apiserver/test_request_encapsulation_middleware.py

```diff
@@ -0,0 +1,298 @@
+)
+def test_unwraps_primitive_values(client, query_path, input_data, expected):
+    """Test unwrapping various primitive types and values."""
+
+    response = client.post(
+        f"/echo?_unwrap={query_path}",
+        json=input_data,
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"received": expected}
+
+
+@pytest.mark.parametrize(
+    "query_path,input_data,expected",
+    [
+        ("/text", {"text": ""}, ""),
+        ("/obj", {"obj": {}}, {}),
+        ("/arr", {"arr": []}, []),
+    ],
+    ids=["empty_string", "empty_object", "empty_array"],
+)
+def test_unwraps_empty_values(client, query_path, input_data, expected):
+    """Test unwrapping empty strings, objects, and arrays."""
+
+    response = client.post(
+        f"/echo?_unwrap={query_path}",
+        json=input_data,
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"received": expected}
+
+
+@pytest.mark.parametrize(
+    "query_path,input_data",
+    [
+        ("data", {"data": "value"}),  # Invalid JSON Pointer (must start with /)
+        ("/nonexistent", {"data": "value"}),  # Path doesn't exist
+        ("/data/missing", {"data": {"present": "value"}}),  # Nested path doesn't exist
+        ("/items/10", {"items": ["a", "b", "c"]}),  # Array index out of bounds
+    ],
+    ids=["invalid_pointer", "undefined_path", "undefined_nested_path", "array_out_of_bounds"],
+)
+def test_error_on_invalid_paths(client, query_path, input_data):
+    response = client.post(
+        f"/echo?_unwrap={query_path}",
+        json=input_data,
+    )
+
+    assert response.status_code == 400
+    assert "_unwrap is not a valid JSONPointer path" in response.json()["message"]
+
+
+@pytest.mark.parametrize(
+    "unwrap_param,query_param,query_value,input_data,expected",
+    [
+        (
+            "_extract",
+            "_extract",
+            "/inner",
+            {"inner": {"custom": "param"}},
+            {"custom": "param"},
+        ),
+        (
+            "_extract",
+            "_unwrap",
+            "/inner",
+            {"inner": {"data": "value"}},
+            {"inner": {"data": "value"}},
+        ),
+    ],
+    ids=["custom_param_works", "default_param_ignored"],
+)
+def test_custom_unwrap_parameter(unwrap_param, query_param, query_value, input_data, expected):
+    """Test using a custom unwrap parameter name."""
+    app = create_test_app(unwrap_param=unwrap_param)
+    client = TestClient(app)
+
+    response = client.post(
+        f"/echo?{query_param}={query_value}",
+        json=input_data,
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"received": expected}
+
+
+def test_handles_empty_request_body(client):
+    response = client.post(
+        "/echo?_unwrap=/data",
+        data=None,
+        headers={"Content-Type": "application/json"},
+    )
+
+    assert response.status_code == 200
+    assert response.json() == {"received": None}
```

Comment:
So this is a special case in which we ignore the invalid JSONPointer path (since it can't point to anything valid inside an empty body) and call it a success regardless of what _unwrap is set to. I see on line 94 of the middleware where we explicitly return when the body is empty. Just calling out this different behavior from `test_error_on_invalid_paths` due to the body.